src/chain/chain.rs

```diff
@@ -657,6 +657,7 @@ impl Chain {
 
     pub fn find_p2p_node_from_addr(&self, socket_addr: &SocketAddr) -> Option<&P2pNode> {
         self.elders()
+            .chain(self.our_joined_members())
```

Comment:
Why do we need both `elders` and `our_joined_members` instead of just `our_joined_members` ?

Reply:
elders also contain neighbour elders

Reply:
Right. But shouldn't it then be `self.neighbour_elder_nodes().chain(self.our_joined_members())`. Otherwise you are including our elders twice.

Reply:
yes, make sense.